components/event-bus/internal/knative/subscription/controller/knativesubscription/reconcile.go

```diff
@@ -3,87 +3,79 @@ package knativesubscription
+	"github.com/kyma-project/kyma/components/event-bus/internal/knative/util"
 )
 
 const (
 	finalizerName = "subscription.finalizers.kyma-project.io"
 )
 
-type reconciler struct {
-	client   client.Client
-	recorder record.EventRecorder
-	time     util.CurrentTime
-}
-
-// Verify the struct implements reconcile.Reconciler
-var _ reconcile.Reconciler = &reconciler{}
+//Reconciler Knative subscriptions reconciler
+type Reconciler struct {
+	*reconciler.Base
+	subscriptionLister subscriptionlistersv1alpha1.SubscriptionLister
+	kymaEventingClient kymaeventingv1alpha1.EventingV1alpha1Interface
 
-func (r *reconciler) InjectClient(c client.Client) error {
-	r.client = c
-	return nil
+	time util.CurrentTime
 }
 
 // Reconcile reconciles a Kn Subscription object
-func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
-	log.Info("Reconcile: ", "request", request)
-
-	ctx := context.TODO()
-	sub := &evapisv1alpha1.Subscription{}
-	err := r.client.Get(ctx, request.NamespacedName, sub)
-
-	// The Knative Subscription may have been deleted since it was added to the work queue. If so, there is
-	// nothing to be done.
-	if errors.IsNotFound(err) {
-		log.Info("Could not find Knative subscription: ", "err", err)
-		return reconcile.Result{}, nil
-	}
+func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
+	log := logging.FromContext(ctx)
 
+	subscription, err := subscriptionByKey(key, r.subscriptionLister)
 	if err != nil {
-		log.Error(err, "unable to Get Knative subscription object")
-		return reconcile.Result{}, err
+		return err
 	}
 
 	// Modify a copy, not the original.
-	sub = sub.DeepCopy()
+	subscription = subscription.DeepCopy()
 
 	// Reconcile this copy of the EventActivation and then write back any status
 	// updates regardless of whether the reconcile error out.
-	requeue, reconcileErr := r.reconcile(ctx, sub)
+	requeue, reconcileErr := r.reconcile(ctx, subscription)
 	if reconcileErr != nil {
-		log.Error(reconcileErr, "error in reconciling Subscription")
-		r.recorder.Eventf(sub, corev1.EventTypeWarning, "SubscriptionReconcileFailed", "Subscription reconciliation failed: %v", reconcileErr)
+		log.Error("error in reconciling Subscription", zap.Error(reconcileErr))
+		r.Recorder.Eventf(subscription, corev1.EventTypeWarning, "SubscriptionReconcileFailed", "Subscription reconciliation failed: %v", reconcileErr)
 	}
 
-	if updateStatusErr := util.UpdateKnativeSubscription(ctx, r.client, sub); updateStatusErr != nil {
-		log.Error(updateStatusErr, "failed in updating Knative Subscription status")
-		r.recorder.Eventf(sub, corev1.EventTypeWarning, "KnativeSubscriptionReconcileFailed", "Updating Kn subscription status failed: %v", updateStatusErr)
-		return reconcile.Result{}, updateStatusErr
+	if updateStatusErr := util.UpdateKnativeSubscription(r.EventingClientSet.MessagingV1alpha1(), subscription); updateStatusErr != nil {
```

Comment:
The name `updateStatusErr` is misleading, this is not a status update. `err` should be good enough here.